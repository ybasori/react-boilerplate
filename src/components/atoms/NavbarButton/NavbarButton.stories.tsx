import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import NavbarButton from "./NavbarButton";

//ðŸ‘‡ This default export determines where your story goes in the story list
const meta: Meta<typeof NavbarButton> = {
  title: "Atoms/NavbarButton",
  component: NavbarButton,
  args: { onClick: fn() },
};

export default meta;
type Story = StoryObj<typeof NavbarButton>;

export const DefaultButton: Story = {
  args: {
    name: "button",
  },
};
