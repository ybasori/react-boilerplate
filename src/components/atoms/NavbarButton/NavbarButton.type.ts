export interface IProps {
  onClick: () => void;
  name: string;
}
