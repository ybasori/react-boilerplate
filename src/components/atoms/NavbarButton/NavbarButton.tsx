import { IProps } from "./NavbarButton.type";

const NavbarButton: React.FC<IProps> = ({ onClick, name }) => {
  return (
    <>
      <li>
        <span onClick={onClick}>{name}</span>
      </li>
    </>
  );
};

export default NavbarButton;
