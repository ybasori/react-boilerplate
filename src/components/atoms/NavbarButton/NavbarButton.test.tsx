import { render, screen, fireEvent } from "@testing-library/react";
import NavbarButton from "./NavbarButton";

describe("Atoms/NavbarButton", () => {
  test("renders the button with the correct label", () => {
    render(<NavbarButton onClick={() => null} name="Click" />);

    // Assert that the button is in the document and has the correct text
    const buttonElement = screen.getByText("Click");
    expect(buttonElement).toBeInTheDocument();
  });

  test("calls the onClick handler when the button is clicked", () => {
    const handleClick = jest.fn();
    render(<NavbarButton onClick={handleClick} name="Click" />);

    const buttonElement = screen.getByText("Click");
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("Snapshot", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <NavbarButton onClick={handleClick} name="Click" />
    );

    expect(container).toMatchSnapshot();
  });
});
