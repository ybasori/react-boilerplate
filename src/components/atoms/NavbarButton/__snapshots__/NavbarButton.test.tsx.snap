// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/NavbarButton Snapshot 1`] = `
<div>
  <li>
    <span>
      Click
    </span>
  </li>
</div>
`;
